core/app_shared.go

```diff
@@ -152,6 +152,37 @@ func (a appShared) hasAttendedEvent(orgID string, appID string, eventID string,
 	return false, nil
 }
 
+func (a appShared) getUserData(orgID string, appID string, userID *string) (*model.UserData, error) {
```

Comment:
As we can have more data in the future as general the approach of getting this info is to become asynchronously. Once all data has been loaded then you are ready to return the result.